plugins/PostProcessingPlugin/scripts/AddCoolingProfile.py

```diff
@@ -0,0 +1,803 @@
+            fan_sp_skirt = check_feature.feature_checker(self.getSettingValueByKey("fan_skirt"), fan_mode)
+            fan_sp_wall_inner = check_feature.feature_checker(self.getSettingValueByKey("fan_wall_inner"), fan_mode)
+            fan_sp_wall_outer = check_feature.feature_checker(self.getSettingValueByKey("fan_wall_outer"), fan_mode)
+            fan_sp_fill = check_feature.feature_checker(self.getSettingValueByKey("fan_fill"), fan_mode)
+            fan_sp_skin = check_feature.feature_checker(self.getSettingValueByKey("fan_skin"), fan_mode)
+            fan_sp_support = check_feature.feature_checker(self.getSettingValueByKey("fan_support"), fan_mode)
+            fan_sp_support_interface = check_feature.feature_checker(self.getSettingValueByKey("fan_support_interface"), fan_mode)
+            fan_sp_prime_tower = check_feature.feature_checker(self.getSettingValueByKey("fan_prime_tower"), fan_mode)
+            fan_sp_bridge = check_feature.feature_checker(self.getSettingValueByKey("fan_bridge"), fan_mode)
+            fan_sp_feature_final = check_feature.feature_checker(self.getSettingValueByKey("fan_feature_final"), fan_mode)
+            fan_combing = self.getSettingValueByKey("fan_combing")
+            if the_end_layer > -1 and by_layer_or_feature == "by_feature":
+                the_end_is_enabled = True #Required so the final speed input can be determined
+            else:
+                the_end_is_enabled = False #There is no ending layer so do the whole file.
+            if the_end_layer == -1 or the_end_is_enabled == False:
+                the_end_layer = 9999999999
+#Find the Layer0Index and the RaftIndex
+        raft_start_index = 0
+        number_of_raft_layers = 0
+        layer_0_index = 0
+        for l_num in range(1,15,1): #Catch the number of raft layers.
+            layer = data[l_num] 
+            if layer.startswith(";LAYER:-"):
+                number_of_raft_layers += 1
+                if raft_start_index == 0:
+                    raft_start_index = l_num
+            if layer.startswith(";LAYER:0"):
+                layer_0_index = l_num
+                break
+
+#Is this a single extruder print on a multi-extruder printer? - get the correct fan number for the extruder being used.
+        if is_multi_fan:
+            T0_used = False
+            T1_used = False
+            T2_used = False
+            T3_used = False
+            for num in range(1,len(data)-1,1): #Bypass the file header and ending gcode.
+                lines = data[num]
+                if "T0" in lines:
+                    T0_used = True
+                if "T1" in lines:
+                    T1_used = True
+                if "T2" in lines:
+                    T2_used = True
+                if "T3" in lines:
+                    T3_used = True
+            if T0_used and (T1_used or T2_used or T3_used):
```

Comment:
... besides the (maybe?) obvious, it would allow to just count the number of extruders used at the end, then set the `is_multi_extr_print` to the right value based on that.